### PR TITLE
NS-230, Nessie front-end (admin console) in Vue.js – a work in progress

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/NS-230

Steps used to transplant the Kelpie proof-of-concept:

* provision front-end with Vue CLI 3
* Transplant important bits from https://github.com/johncrossman/kelpie-vue
* Get linters to pass
* Load homepage to get Nessie `/api/schedule`

The UX is far from done and the auth is not wired up. To get this off the ground I want to see Travis pass. Then we'll move on to making it all real.

### CODE REVIEW

File: src/components/Job.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  data() {
+    return {
+      job: {},
+      jobStatus: null,
+    };
+  },
+  methods: {
+    getJob(jobId) {
+      axios.get(`http://localhost:5000/api/schedule/${jobId}`).then((response) => {
```
Comment (johncrossman) [inline src/components/Job.vue]: I'll amend this old Kelpie URI out.

**Comment by paulkerschen** [comment]
@johncrossman Travis might be running into this: https://github.com/vuejs/vue-cli/issues/1105

**Comment by paulkerschen** [comment]
Sorry, _now_ it looks like Python 2 vs 3 incompatibility with node-gyp?

https://github.com/nodejs/node-gyp/issues/193

Something in vue, possibly far down, is perhaps requiring node-gyp?

**Comment by paulkerschen** [comment]
Specifically I think pytest has to be told not to load the node-modules dir.

https://stackoverflow.com/questions/47458933/getting-pytest-to-run-in-a-django-cookiecutter-python-3-x-project

**Comment by raydavis** [comment]
I had no idea pytest was so adventurous!

**Comment by paulkerschen** [comment]
Thanks @johncrossman - look forward to checking out this room with a vue.

**Pull request merged**